libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -179,30 +637,116 @@ void SpvcIrPass::GenerateSpirvCrossIR(Instruction *inst) {
       break;
     }
 
+    // opcode: 253
     case SpvOpReturn: {
       assert(current_block_ && "Trying to end a non-existing block.");
-      // TODO (sarahM0): refactor this into TerminateBlock( ... terminator type
-      // ...), which also resets current_block_ to nullptr. Once having one more
-      // terminator case.
+      // TODO (sarahM0): refactor this into TerminateBlock( ... terminator
+      // type
+      // ...), which also resets current_block_ to nullptr. Once having one
+      // more terminator case.
 
       current_block_->terminator = spirv_cross::SPIRBlock::Return;
       current_block_ = nullptr;
       break;
     }
 
-    case SpvOpFunctionEnd: {
-      // Very specific error message, but seems to come up quite often.
-      assert(!current_block_ &&
-             "Cannot end a function before ending the current block.\n"
-             "Likely cause: If this SPIR-V was created from glslang HLSL, "
-             "make sure the entry point is valid.");
-      current_function_ = nullptr;
+    // opcode: 52
+    case SpvOpSpecConstantOp: {
+      assert(inst->NumOperandWords() >= 3 &&
+             "OpSpecConstantOp not enough arguments.");
+
+      uint32_t result_type = inst->type_id();
+      uint32_t id = inst->result_id();
+      auto spec_op = static_cast<spv::Op>(inst->GetSingleWordInOperand(0u));
+
+      if (inst->NumOperandWords() - 3 > 0) {
+        uint32_t *args = new uint32_t[inst->NumOperandWords() - 3];
+        for (uint32_t i = 0u; i < inst->NumOperandWords() - 3; i++) {
+          args[i] = inst->GetSingleWordInOperand(i + 1);
+        }
+        set<spirv_cross::SPIRConstantOp>(id, result_type, spec_op, args,
+                                         inst->NumOperandWords() - 3);
+      }
       break;
     }
 
-    default:
+    // TODO(sarahM0): These opcodes are processed in the generator.
```

Comment:
There should probably be an issue on github tracking this.

Reply:
issue #854